experimental/casemapping/src/greek_to_me/mod.rs

```diff
@@ -0,0 +1,253 @@
+    pub(crate) fn is_greek_letter(self) -> bool {
+        self.uppercase.is_some()
+    }
+}
+
+impl From<u16> for GreekPrecomposedLetterData {
+    fn from(sixteen: u16) -> Self {
+        let mut this = Self::default();
+        if sixteen & (1 << 8) != 0 {
+            let int = u32::from(sixteen & 0xFF);
+            let ch = int + GREEK_BLOCK_START;
+            let ch = char::try_from(ch);
+            debug_assert!(ch.is_ok());
+            this.uppercase = ch.ok();
+        } else {
+            debug_assert!((sixteen & 0xFF00) == 0)
+        }
+        this.ypogegrammeni = sixteen & (1 << 9) != 0;
+        this.dialytika = sixteen & (1 << 10) != 0;
+        this.accented = sixteen & (1 << 11) != 0;
+        debug_assert!(
+            sixteen & (0xFFFF - (1 << 12) + 1) == 0,
+            "Bits 12..16 should not be set in GreekPrecomposedLetterData"
+        );
+
+        this
+    }
+}
+
+impl From<GreekPrecomposedLetterData> for u16 {
+    fn from(this: GreekPrecomposedLetterData) -> Self {
+        let mut sixteen = 0;
+        if let Some(uppercase) = this.uppercase {
+            let delta = u32::from(uppercase) - GREEK_BLOCK_START;
+            debug_assert!(delta <= 0xFF);
+            sixteen = u16::try_from(delta).unwrap_or(0);
+
+            sixteen |= 1 << 8;
+        }
+        if this.ypogegrammeni {
+            sixteen |= 1 << 9;
+        }
+        if this.dialytika {
+            sixteen |= 1 << 10;
+        }
+        if this.accented {
+            sixteen |= 1 << 11;
+        }
+        sixteen
+    }
+}
+
+impl From<GreekPrecomposedLetterData> for u32 {
+    fn from(this: GreekPrecomposedLetterData) -> Self {
+        u16::from(this).into()
+    }
+}
+
+/// General diacritic information for a character or string
+#[derive(Copy, Clone, Default, PartialEq, Eq, Debug)]
+pub struct GreekDiacritics {
+    /// Whether it has an accent
+    pub accented: bool,
+    /// Whether it has a dialytika
+    pub dialytika: bool,
+    /// Whether it contains an ypogegrammeni (precomposed)
+    pub precomposed_ypogegrammeni: bool,
+    /// Whether it contains an ypogegrammeni (combining)
+    pub combining_ypogegrammeni: bool,
+}
+
+pub const DIALYTIKA: char = '\u{0308}';
+pub const DIALYTIKA_TONOS: char = '\u{0344}';
+pub const YPOGEGRAMMENI: char = '\u{0345}';
+
+#[macro_export]
+#[doc(hidden)]
+macro_rules! accent_marks {
+    () => {
+        // Accent marks that map to a tonos when found with a lone eta
+        // https://util.unicode.org/UnicodeJsps/list-unicodeset.jsp?a=%5B%5Cu0300+%5Cu0301+%5Cu0342+%5Cu0302+%5Cu0303+%5Cu0311%5D&g=&i=
+        '\u{0300}' | '\u{0301}' | '\u{0342}' | '\u{0302}' | '\u{0303}' | '\u{0311}'
+    };
+    (extra) => {
+        // Other combining marks that are expected to be used with Greek
+        // https://util.unicode.org/UnicodeJsps/list-unicodeset.jsp?a=%5B%5Cu0304+%5Cu0306+%5Cu0313+%5Cu0314+%5Cu0343%5D&g=&i=
+        '\u{0304}' | '\u{0306}' | '\u{0313}' | '\u{0314}' | '\u{0343}'
+    };
+}
```

Comment:
Those two categories are accents (and accent lookalikes, really), and non-accent diacritics (these are breathings and vowel length diacritics). Let’s not call the latter accents, this is hopelessly confusing.